### PR TITLE
test: add hydration parachain regression tests

### PR DESCRIPTION
**Pull request by dmoka**
We added our first regressions tests using the provided framework.

### CODE REVIEW

File: packages/polkadot/src/__snapshots__/hydration.moonbeam.test.ts.snap [hunk 1]
```diff
@@ -0,0 +1,1237 @@
...
+  "reserved": 0,
+}
+`;
+
+exports[`hydration & moonbeam > hydration transfer GLMR to moonbeam > balance on to chain 1`] = `
+{
+  "consumers": 0,
+  "data": {
+    "flags": 0,
+    "free": 0,
```
Comment (xlc) [inline packages/polkadot/src/__snapshots__/hydration.moonbeam.test.ts.snap:849]: is this expected?

File: packages/polkadot/src/__snapshots__/hydration.moonbeam.test.ts.snap [hunk 2]
```diff
@@ -0,0 +1,1237 @@
+// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html
+
+exports[`hydradation & moonbeam > hydradation transfer DOT to moonbeam > balance on from chain 1`] = `
```
Comment (xlc) [inline packages/polkadot/src/__snapshots__/hydration.moonbeam.test.ts.snap:3]: looks like you have some old snapshots. try delete this file and regenerate

**Comment by xlc** [comment]
@dmoka do you want to finish this?

**Comment by dmoka** [comment]
> @dmoka do you want to finish this?

Absolutely, thanks for the patience, I had some higher prior stuff to work on last week. Early next week I will look into this and wrap it up

File: packages/polkadot/src/__snapshots__/hydration.moonbeam.test.ts.snap (hunk 1 above)
Comment (dmoka) [reply packages/polkadot/src/__snapshots__/hydration.moonbeam.test.ts.snap:849]: You are right, the transaction indeed fails, and the asset is trapped on Moonbeam.

Is there a way to enable detailed xcm logs so I could troubleshoot the issue easily? When we use chopsticks, we usually use `runtime-log-level: 5` in the runtime yaml config file, but in this repo I could not find the same, only to increase log level with `LOG_LEVEL=debug`, but that does not lead to detailed xcm runtime logs.

**Comment by dmoka** [comment]
@xlc For one of our tests ('moonbeam transfer DOT to hydration'), we would need to create a new block on the `toChain`.

But in the `runXtokenstHorizontal` method, [new block is only crated in the `fromChain`](https://github.com/open-web3-stack/polkadot-ecosystem-tests/blob/4a099e35ea7ea3c87849e82a72ca31646561c755/packages/shared/src/xcm/runXtokenstHorizontal.ts#L46). 

As a solution, we could add a new param to `runXtokenstHorizontal`, with a function type, with name `postAction` (or so) that we could easily execute there.

What do you think? Or you might have a better solution in mind?

File: packages/polkadot/src/__snapshots__/hydration.moonbeam.test.ts.snap (hunk 1 above)
Comment (xlc) [reply packages/polkadot/src/__snapshots__/hydration.moonbeam.test.ts.snap]: use `RUNTIME_LOG_LEVEL=5`

https://github.com/open-web3-stack/polkadot-ecosystem-tests/blob/master/README.md#environment-variables

**Comment by xlc** [comment]
Chopsticks should automatically create new blocks on a chain when it receives an incoming XCM so you shouldn't need to manually create new block unless you need an additional block beyond the automatically created one.
What is the reason you need it? For example if it is the auto new block for XCM isn't working well, that's something we can fix in Chopsticks.

**Comment by dmoka** [comment]
> Chopsticks should automatically create new blocks on a chain when it receives an incoming XCM so you shouldn't need to manually create new block unless you need an additional block beyond the automatically created one. What is the reason you need it? For example if it is the auto new block for XCM isn't working well, that's something we can fix in Chopsticks.

Hmm, now running it without using `toChain.newBlock` works well, the from and tochain balance checks are looking fine now. If you could double check it, that would be appreciated, otherwise LGTM from my point of view, thanks

**Review by xlc — APPROVED** [review]

**Pull request merged**